lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -772,8 +794,22 @@ def alerts(self) -> dict:
             if not alert_rules:
                 continue
 
+            transformation_provider = _TransformationProvider(self._charm)
+            scrape_metadata = json.loads(relation.data[relation.app].get("scrape_metadata", "{}"))
+
             try:
                 for group in alert_rules["groups"]:
+                    rules = group.get("rules", [])
+
+                    for rule in rules:
+                        expression = rule["expr"]
+                        rule["expr"] = transformation_provider.apply_label_matcher(
+                            expression,
+                            JujuTopology.from_relation_data(
+                                scrape_metadata
+                            ).as_dict_with_promql_labels(),
+                        )
```

Comment:
I noticed that the following line in `load_alert_rule_from_file()` was not removed as part of this PR:
```python
rule["expr"] = topology.render(rule["expr"])
```
I'm not sure why. Also all the example rule files should have `%%juju...%%` removed, no?

Reply:
Yeah, it does nothing if the templating string is not present, but I kept it for backwards-compatibility with existing rules. @mmanciop is this an OK breaking change for you given that we're still in the `edge` channel?

Reply:
Regarding removing the template strings from, that would break the topology injection currently as we can't attach resources during testing. I could mock away the apply_label_selectors function and make it return the correct values though. I'll ping you once I have it in 👍🏽

Reply:
Tests did not check for that, so nothing had to be changed, but kept one with topology template labels to make sure we have that present until we make a decision on whether to keep it around.